source/client-side-encryption/tests/unified/createKey.yml

```diff
@@ -131,3 +131,100 @@ tests:
         expectResult:
           - masterKey:
               provider: local
+
+  - description: create datakey with no keyAltName
+    operations:
+      - name: createKey
+        object: *clientEncryption0
+        arguments:
+          kmsProvider: local
+          opts:
+            keyAltNames: []
+        expectResult: { $$type: binData }
+      - name: find
+        object: *collection0
+        arguments:
+          filter: {}
+        expectResult:
+          - _id: { $$type: binData }
+            # keyAltNames field should not exist if no keyAltNames are given.
+            keyAltNames: { $$exists: false }
+            keyMaterial: { $$type: binData }
+            creationDate: { $$type: date }
+            updateDate: { $$type: date }
+            status: { $$type: int }
+            masterKey: { $$type: object }
+
+  - description: create datakey with single keyAltName
+    operations:
+      - name: createKey
+        object: *clientEncryption0
+        arguments:
+          kmsProvider: local
+          opts:
+            keyAltNames: ["local_key"]
+        expectResult: { $$type: binData }
```

Comment:
I'm not sure it'd be helpful/relevant here, but I just realized that there's no way to assert the subtype of a binary value. The server's [`$type`](https://www.mongodb.com/docs/manual/reference/operator/query/type/) operator doesn't support this, but I suppose we could in the Unified Test Format if there was a need. If so, feel free to log a ticket to throw in the backlog.

If so, you could always add some `# TODO` comments to these tests in places where you would have liked to use the functionality.

Reply:
Ticket created: https://jira.mongodb.org/browse/DRIVERS-2336